### ISSUE TITLE
Bug: Regression between 17.15.1 and 17.16.1 with file-extension-in-import

### ISSUE THREAD

**Issue by Zamiell** [eslint-community/eslint-plugin-n#413]
## What rule do you want to report?

file-extension-in-import

## Description

There is a regression in the latest version such that I have code like this:

```ts
import { $ } from "complete-node";
```

Now triggers the rule:

```
require file extension '.js'  n/file-extension-in-import
```

Which is a false positive, because this is a package import from a dependency, it's not referring to a file. However in this case I am using tsconfig paths like this:

```
{
  "compilerOptions": {
    "paths": {
      "complete-node": ["./packages/complete-node/src/index.ts"],
    },
  },
}
```

So I think it is getting confused because "index.ts" is a file. Anyways, it must be related to something in the last update.

## Steps to Reproduce

```sh
git clone git@github.com:complete-ts/complete.git
cd complete
git switch n-bug
npm ci
cd packages/complete-cli
npx eslint src/constants.ts
```

## What did you expect to happen?

To have no ESLint warning/error occur.

## Environment

Node version: v22.14.0
npm version: 10.9.2
ESLint version: 9.21.0
eslint-plugin-n version: 17.16.1
Operating System: Windows

## Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by rosskevin** [comment]
I confirm that `17.5.1` was working correctly in a monorepo with interdependent packages, and now `17.16.2`  I added resolutions to override, and our public monorep is linting again.  https://github.com/alienfast/logger/commit/5cef302134b97ca54841bfbf533dec67b5c3c2f5

If it is related to tsconfigs, we are using `project references` e.g. `"references": [{ "path": "../logger" }]`

**Comment by ota-meshi** [comment]
I think using v17.17 will solve that problem, could you give it a try?

**Comment by Zamiell** [comment]
Yes, that looks like it fixed it, thank you @ota-meshi !
